server/src/test/java/com/vaadin/data/provider/DataCommunicatorTest.java

```diff
@@ -322,4 +322,12 @@ public void requestTooMuchRowsFail() {
         communicator.onRequestRows(0, communicator.getMaximumAllowedRows() + 10,
                 0, 0);
     }
+
+    @Test
+    public void requestTooMuchRowsOverride() {
+        TestDataCommunicator communicator = new TestDataCommunicator();
```

Comment:
I think this should be tested with an actual Component to make sure the new limit can used for the first load